src/server/plugins/engine/pageControllers/PageControllerBase.ts

```diff
@@ -389,7 +391,32 @@ export class PageControllerBase {
 
       // Iterate all components on this page and pull out the saved values from the state
       for (const component of nextPage.components.items) {
-        newValue[component.name] = currentState[component.name]
+        let componentState = currentState[component.name]
+
+        // For evaluation context purposes, optional `CheckboxesField`s
+        // with an undefined value (i.e. nothing selected) should default to [].
+        // This way conditions are not evaluated against `undefined` which throws errors (see LN234).
+        // Currently these errors are caught and the evaluation returns default `false`.
+        // (See src/server/plugins/engine/models/FormModel.ts LN227)
+        // For negative conditions this is a problem because E.g.
+        // The condition: 'selectedchecks' does not contain 'someval'
+        // should return true IF 'selectedchecks' is undefined, not throw and return false.
+        // Similarly for optional `RadiosField`, the evaluation context should default to null.
```

Comment:
Thanks @davidjamesstone, this write up is really clear

Mind if we use JSDoc `@link` to replace line numbers that can drift?

```suggestion
        /**
         * For evaluation context purposes, optional {@link CheckboxesField}
         * with an undefined value (i.e. nothing selected) should default to [].
         * This way conditions are not evaluated against `undefined` which throws errors.
         * Currently these errors are caught and the evaluation returns default `false`.
         * @see {@link PageControllerBase.getNextPage} for `undefined` return value
         * @see {@link FormModel.makeCondition} for try/catch block with default `false`
         * For negative conditions this is a problem because E.g.
         * The condition: 'selectedchecks' does not contain 'someval'
         * should return true IF 'selectedchecks' is undefined, not throw and return false.
         * Similarly for optional {@link RadiosField}, the evaluation context should default to null.
         */
```

Reply:
[Resolved](https://github.com/DEFRA/forms-runner/pull/65/commits/7af8bdd5f691ef2e08294784b2a6a0c5380a7c12)